src/client.ts

```diff
@@ -1436,7 +1449,33 @@ export class StreamChat<
       ReactionType
     >({ client: this });
 
-    return await this.wsConnection.connect();
+    try {
+      // if WSFallback is enabled, ws connect should timeout faster so fallback can try
+      return await this.wsConnection.connect(
+        this.options.enableWSFallback ? this.defaultWSTimeoutWithFallback : this.defaultWSTimeout,
+      );
+    } catch (err) {
+      // run fallback only if it's WS/Network error and not a normal API error
+      // make sure browser is online before even trying the longpoll
+      if (this.options.enableWSFallback && isWSFailure(err) && isOnline()) {
```

Comment:
I wonder if we shouldn't remove thhe online check. If you're offline the `this.wsConnection.connect` fails and `isWsFailure` will return false anyways. On the other hand, if it could be the case that you have a wsFailure and you're offline, in RN this check wouldn't work, and we would trigger the fallback while offline. I'm not sure if I'm missing something, though. Does it make sense?

Reply:
Following up on this: so I just tested on RN. If you access `window.navigator` it looks like following:

```
{"product": "ReactNative"}
```

And thus `isOnline()` function will return always `undefined` -> fallback will never be executed. Maybe we can provide a way so that RN SDK can let JS client know about the online status (e.g., we can provide js client with async function to fetch online status).